contracts/protocol/facets/SellerHandlerFacet.sol

```diff
@@ -94,73 +94,178 @@ contract SellerHandlerFacet is SellerBase {
+                // If admin address exists, admin address owner must approve the update for prevent front-running
+                lookups.sellerPendingUpdates[_seller.id].admin = _seller.admin;
+                needsApproval = true;
+            }
+        }
+
+        if (seller.treasury != _seller.treasury) {
+            require(_seller.treasury != address(0), INVALID_ADDRESS);
+            // Update treasury
+            seller.treasury = _seller.treasury;
+        }
+
+        if (_seller.operator != seller.operator) {
+            require(_seller.operator != address(0), INVALID_ADDRESS);
+            // Operator address owner must approve the update for prevent front-running
+            lookups.sellerPendingUpdates[_seller.id].operator = _seller.operator;
+            needsApproval = true;
+        }
+
+        if (_seller.clerk != seller.clerk) {
+            require(_seller.clerk != address(0), INVALID_ADDRESS);
+            // Clerk address owner must approve the update for prevent front-running
+            lookups.sellerPendingUpdates[_seller.id].clerk = _seller.clerk;
+            needsApproval = true;
+        }
+
+        if (needsApproval) {
+            emit SellerUpdateRolesRequested(_seller.id, _seller, sender);
+        }
+
+        // Notify watchers of state change
+        emit SellerUpdated(_seller.id, seller, _authToken, sender);
+    }
+
+    function approveSellerUpdate(uint256 _sellerId) external {
+        // Cache protocol lookups for reference
+        ProtocolLib.ProtocolLookups storage lookups = protocolLookups();
+        address sender = msgSender();
+
+        // Get seller pending update
+        Seller storage _sellerPendingUpdate = lookups.sellerPendingUpdates[_sellerId];
+
+        require(
+            _sellerPendingUpdate.admin != address(0) ||
+                _sellerPendingUpdate.operator != address(0) ||
+                _sellerPendingUpdate.clerk != address(0),
+            NO_PENDING_SELLER_ROLE_UPDATE
+        );
+
+        sellerPreUpdateChecks(_sellerId, _sellerPendingUpdate.admin, lookups);
+        sellerPreUpdateChecks(_sellerId, _sellerPendingUpdate.operator, lookups);
+        sellerPreUpdateChecks(_sellerId, _sellerPendingUpdate.clerk, lookups);
+
+        // Get storage location for seller
+        (, Seller storage seller, ) = fetchSeller(_sellerId);
+
+        // Approve operator address
+        if (_sellerPendingUpdate.operator == sender) {
+            // If operator changed, update the operator and transfer the ownership of NFT voucher
+            if (seller.operator != _sellerPendingUpdate.operator) {
+                delete lookups.sellerIdByOperator[seller.operator];
+
+                // Update operator
+                seller.operator = _sellerPendingUpdate.operator;
+
+                // Transfer ownership of NFT voucher to new operator
+                IBosonVoucher(lookups.cloneAddress[seller.id]).transferOwnership(seller.operator);
```

Comment:
```suggestion
                IBosonVoucher(lookups.cloneAddress[_sellerId]).transferOwnership(seller.operator);
```

Reply:
I think ` sender` or ` _sellerPendingUpdate.operator` could also be used as the parameter to `transerOwnership` because passing `seller.operator` causes an unnecessary SLOAD. To me it also makes it clearer what is happening.
Whichever variable us used should also be used in line 204 as the key. Right now there are several different variables being used that should have the same value.

Reply:
`_sellerPendingUpdate.operator` also reads from storage, so I suppose we should just use `sender`.

Reply:
fixed on both places